src/main/webapp/script.js

```diff
@@ -38,8 +48,41 @@ function displayError() {
 }
 
 function getZipCode() {
-  let zip = document.getElementById('zipCode').value;
-  fetchList('/data?zipCode=' + zip);
+  hideEntryContainer();
+  initiateLoaderCircle();
+  zip = document.getElementById('entryZipCode').value;
+  document.getElementById('zipCode').value = zip;
+  locationQuery = '/data?zipCode=' + zip;
+  product = document.getElementById('product').value;
+  if (product === '') {
+    fetchList(locationQuery + '&product=NULL');
+  }
+  else {
```

Comment:
Same comment as above. Also, given this is the same logic as above, perhaps move it to a different method, and call that where relevant.

Reply:
I refactored this bit of code into a separate function (fetchByQueryString()).